### PR TITLE
Fix compile error in Macchiato.h

### PR DESCRIPTION
**Pull request by Alex-Tsvetanov**
`Macchiato.h` can't be compiled, because 
```
In file included from example2.cpp:2:0:
Macchiato.h: In function ‘void Macchiato::MacchiatoParseCLIArgs(int, char* const*)’:
Macchiato.h:453:35: error: ‘strcmp’ was not declared in this scope
    if(strcmp(argv[i], "--no-color") == 0) {
                                   ^
```
And it can be easily fixed by including `<cstring>` into `Macchiato.h`.

### CODE REVIEW

**Comment by MadLittleMods** [comment]
@Alex-Tsvetanov What target/platform, VS/Arduino, etc are you seeing the compile error in?

Essentially wondering why it worked for me in my testing but not in your environment

File: include/Macchiato.h [hunk 1]
```diff
@@ -20,6 +20,9 @@
 // fabs
 #include <cmath>
 
+//strcmp
+#include <cstring>
+
```
Comment (MadLittleMods) [inline include/Macchiato.h]: We probably want to add this to the `single_include` version as well

**Review by MadLittleMods — CHANGES_REQUESTED** [review]

**Comment by Alex-Tsvetanov** [comment]
I compiled it on Manjaro Linux 4.4.33-1 with:
```
$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.1/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc-multilib/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --enable-libmpx --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release
Thread model: posix
gcc version 6.3.1 20170306 (GCC) 
```
by this command line ``g++ example2.cpp -std=c++17 -O2 -Wall -o test``

But in C++ ``#include <cstring>`` is neccessary when you use ``strcmp(char*, char*)`` because it is more correct than otherwise.

By standart when you use any function from library, which is not included, you should include it!

File: include/Macchiato.h (hunk 1 above)
Comment (Alex-Tsvetanov) [reply include/Macchiato.h]: Yes, you should add this line to the ``single_include`` as well.

File: examples/arduino_example1.ino [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-			it("should qux", [&]() {
-				return expect(false).to->equal(true)->getResult();
-			});
-
-			// An `it` call without a callback is considered "pending"
-			// In other words, the test still needs to written/implemented.
-			it("should norf");
-		});
-	});
-});
```
Comment (MadLittleMods) [inline examples/arduino_example1.ino:23]: Please don't get rid of the examples 😕

File: examples/arduino_example2.ino [hunk 3]
```diff
@@ -1,44 +0,0 @@
...
-				// In other words, the test still needs to written/implemented.
-				it("should norf");
-			});
-		});
-
-		Serial.print(Macchiato::GetResultantTestOutput().value);
-
-		sinceSerial = 0;
-	}
-}
```
Comment (MadLittleMods) [inline examples/arduino_example2.ino:44]: Please don't get rid of the examples 😕

File: examples/example2.cpp [hunk 4]
```diff
@@ -3,25 +3,35 @@
...
+			return expect(ProgramPrototype.bar()).to->equal(999)->getResult();
 		});
 	});
 
 	std::cout << Macchiato::GetResultantTestOutput() << std::endl;
 
 	return 0;
-}
\ No newline at end of file
+}
```
Comment (MadLittleMods) [inline examples/example2.cpp:37]: I don't think this adds more value over the simple version that shows off a pass, fail, pending as well.

File: include/Macchiato.h [hunk 5]
```diff
@@ -19,6 +19,8 @@
 
 // fabs
 #include <cmath>
+//strcmp
+#include <cstring>
```
Comment (MadLittleMods) [inline include/Macchiato.h:23]: We need this in the `single_include` version as well

**Pull request closed without merging**